File: Web-Frontend/src/api/authApi.js
import API from "./axios";

export const loginUser = (data) => API.post("/auth/login", data);

export const forgotPassword = (email) =>
  API.post("/auth/forgot-password", { email });

export const resetPassword = (data) => API.post("/auth/reset-password", data);
export const getMe = () => API.get("/auth/me");

File: Mobile-App/mint_mobile/lib/api/auth_api.dart
import 'package:dio/dio.dart';
import 'dio_client.dart';

class AuthApi {
  static Future<Response> login(String email, String password) {
    return DioClient.instance.post('/auth/login', data: {
      'email': email,
      'password': password,
    });
  }

  static Future<Response> getMe() {
    return DioClient.instance.get('/auth/me');
  }

  static Future<Response> forgotPassword(String email) {
    return DioClient.instance.post('/auth/forgot-password', data: {'email': email});
  }

  static Future<Response> resetPassword(String token, String password) {
    return DioClient.instance.post('/auth/reset-password', data: {
      'token': token,
      'password': password,
    });
  }
}
